### PR TITLE
[WIP]according to APR audit.

### PR DESCRIPTION
**Pull request by ilanDoron**
limit some configuration parameters.
and overflow checks.

### CODE REVIEW

File: contracts/reserves/aprConversionRate/LiquidityFormula.sol [hunk 1]
```diff
@@ -147,6 +150,9 @@ contract LiquidityFormula is UtilMath {
     {
         uint pe = pE(r, pMIn, e, precision);
         uint rpe = r * pe;
+
+        require(!checkMultOverflow(rpe, deltaT));
+        require(precision * precision + rpe * deltaT/precision > precision * precision);
```
Comment (manhlx3006) [inline contracts/reserves/aprConversionRate/LiquidityFormula.sol]: In line 163, we must check r to be non-zero before using `/`

Comment (manhlx3006) [reply contracts/reserves/aprConversionRate/LiquidityFormula.sol]: Similar to in func `countLeadingZeros`, there is `if (p/(q*denomator) > 0) return uint(i);` which also need to check `q` to be non-zero.

Comment (ilanDoron) [reply contracts/reserves/aprConversionRate/LiquidityFormula.sol]: r in FP is checked to be > 0 when set.
and later it is never modified.

Comment (ilanDoron) [reply contracts/reserves/aprConversionRate/LiquidityFormula.sol]: q is set to be precision * precision.
and precision parameter is set on and can't be zero (after my changes)

Comment (manhlx3006) [reply contracts/reserves/aprConversionRate/LiquidityFormula.sol]: but they are public call right? so we can call with 0 value unless it is internal func

**Comment by ilanDoron** [comment]
this PR only for code compare.
APR will be changed only after final APR report

**Pull request closed without merging**